lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -290,3 +297,363 @@ class CourseGradingPolicy(CourseViewMixin, ListAPIView):
+            Returned only if the "block_count" input parameter contains this block's type.
+
+          * block_json: (dict) The JSON data for this block.
+            Returned only if the "block_json" input parameter contains this block's type.
+
+          * block_url: (string) The URL to retrieve the HTML rendering of this block.  The HTML could include
+            CSS and Javascript code.  This URL can be used as a fallback if the custom block_json for this
+            block type is not requested and not supported.
+
+          * web_url: (string) The URL to the website location of this block.  This URL can be used as a further
+            fallback if the block_url and the block_json is not supported.
+
+          * graded (boolean) Whether or not the block or any of its descendants is graded.
+            Returned only if "graded" is included in the "fields" parameter.
+
+          * format: (string) The assignment type of the block.
+            Possible values can be "Homework", "Lab", "Midterm Exam", and "Final Exam".
+            Returned only if "format" is included in the "fields" parameter.
+
+          * responsive_ui: (boolean) Whether or not the block's rendering obtained via block_url is responsive.
+            Returned only if "responsive_ui" is included in the "fields" parameter.
+
+        * navigation: A dictionary that maps block IDs to a collection of navigation information about each block.
+          Each block contains the following fields.
+
+          * descendants: (list) A list of IDs of the children of the block if the block's depth in the
+            course hierarchy is less than the navigation_depth.  Otherwise, a list of IDs of the aggregate descendants
+            of the block.
+
+        * blocks_navigation: A dictionary that combines both the blocks and navigation data.
+
+    """
+    class RequestInfo(object):
+        """
+        A class for encapsulating the request information, including what optional fields are requested.
+        """
+        DEFAULT_FIELDS = "children,graded,format,responsive_ui"
+
+        def __init__(self, request, course):
+            self.request = request
+            self.course = course
+            self.field_data_cache = None
+
+            # check what fields are requested
+            try:
+                # fields
+                self.fields = set(request.GET.get('fields', self.DEFAULT_FIELDS).split(","))
+
+                # children
+                self.children = 'children' in self.fields
+                self.fields.discard('children')
+
+                # block_count
+                self.block_count = request.GET.get('block_count', "")
+                self.block_count = (
+                    self.block_count.split(",") if self.block_count else []
+                )
+
+                # navigation_depth
+                self.navigation_depth = int(request.GET.get('navigation_depth', '3'))
```

Comment:
I updated the docstring to say:
`Default is 3 since typical navigational views of the course show a maximum of chapter->sequential->vertical.`